doc/code_architecture.md

```diff
@@ -0,0 +1,39 @@
+No PAB is used here, we use Cardano-Api with custom helpers
+for transaction construction. On-chain scripts are written
+in pure Plutus.
+
+Utilities:
+
+* `HydraAuction.Plutus.Extras` - extension of `PlutusExtra` package,
+   which contains reimpelementation for important `plutus-ledger` stuff
+* `HydraAuction.Plutus.Orphans` - orphans for Plutus stuff.
+* `HydraAuction.Fixture` bunch of known actors (== Cardano keys) to use.
+   Hydra does have one too, but it only have 3 general-use actors.
+
+Library code:
+
+* `HydraAuction.Types` domain types used by scripts.
+   haskell-language-server-wrapperAnd our `Natural` implementation, lol.
```

Comment:
```suggestion
* `HydraAuction.Types` domain types used by scripts as well as  our own `Natural`
   implementation, because we cannot add the dependency on `plutus-apps` that 
   `plutus-numeric` would require. 
```

Reply:
```suggestion
* `HydraAuction.Types` domain types used by scripts
   as well as  our own `Natural` implementation,
   because we cannot add the dependency on `plutus-apps`
   that `plutus-numeric` would require. 
```